doc/md/guides: Add doc about index operator classes in PostgreSQL (#1300)
ariga/atlas: doc/md/guides/postgres/index-operator-classes.md

 Gather  (cost=1000.00..15874.33 rows=100 width=44) (actual time=5.956..227.339 rows=78 loops=1)
   Workers Planned: 2
   Workers Launched: 2
   ->  Parallel Seq Scan on internet_provider  (cost=0.00..14864.33 rows=42 width=44) (actual time=10.923..211.886 rows=26 loops=3)
         Filter: ((subscriber_name)::text ~~ 'Shirley C%'::text)
         Rows Removed by Filter: 333307
 Planning Time: 4.955 ms
 Execution Time: 228.132 ms
(8 rows)

Time: 243.573 ms
```
Notice that the index `internet_provider_idx` that we created was not used in order to execute this query. Instead, the `Parallel Seq Scan` operation was performed. As a result, the total execution time and cost are still too high.

:::info
In a parallel sequential scan, the table's blocks will be divided into ranges and shared among the cooperating processes. Each worker process will complete the scanning of its given range of blocks before requesting an additional range of blocks. To learn more about Parallel Plans in PostgreSQL, visit the official documentation [here](https://www.postgresql.org/docs/current/parallel-plans.html).
:::

> Now, you might be wondering why the index that we created was not being used in the execution of this query. This is a point where having knowledge about usage of operator classes becomes important. 

Reviewer: ```suggestion
Now, you might be wondering why the index that we created was not used in the execution of this query. This when having knowledge about the usage of operator classes becomes important. 
```

```

```console title="Output"
CREATE INDEX
Time: 2375.380 ms (00:02.375)
```

This time, we specified an operator class `varchar_pattern_ops` in our index definition. `varchar_pattern_ops` is a built-in operator class which supports B-tree indexes on the data-type `varchar`. Let’s check the performance and plan of the query we previously used with the following command:

```sql
EXPLAIN ANALYZE
SELECT
        *
FROM    
        internet_provider
WHERE
        subscriber_name LIKE 'Shirley C%'
```
```console title="Output"
                                                                 QUERY PLAN                                                                  
---------------------------------------------------------------------------------------------------------------------------------------------
 Index Scan using internet_provider_idx on internet_provider  (cost=0.42..8.45 rows=100 width=44) (actual time=0.025..0.131 rows=78 loops=1)
   Index Cond: (((subscriber_name)::text ~>=~ 'Shirley C'::text) AND ((subscriber_name)::text ~<~ 'Shirley D'::text))
   Filter: ((subscriber_name)::text ~~ 'Shirley C%'::text)
 Planning Time: 0.103 ms
 Execution Time: 0.153 ms
(5 rows)
```

> Amazing! This time, Index Scan was performed using `internet_provider_idx`. As a result the cost, planning time, and execution time for our query have reduced significantly, as we expected. 

Reviewer: ```suggestion
Amazing! This time, Index Scan was performed using `internet_provider_idx`. As a result, the cost, planning time, and execution time for our query have reduced significantly, as we expected. 
```

:::info
If you are just getting started, install the latest version of Atlas using the guide to [setting up Atlas](https://atlasgo.io/cli/getting-started/setting-up).
:::

#### Managing Operator Classes with Atlas

We will first use the `atlas schema inspect` command to get an [HCL](https://atlasgo.io/guides/ddl#hcl) representation of the table we created earlier (without any indexes other than primary index on `id` column) by using the Atlas CLI:

```console
atlas schema inspect -u "postgres://postgres:mysecretpassword@localhost:5432/internet_provider_db?sslmode=disable" > schema.hcl
```
```hcl title="schema.hcl"
table "internet_provider" {
  schema = schema.public
  column "id" {
    null = false
    type = serial
  }
  column "subscriber_name" {
    null = false
    type = varchar(255)
  }
  column "email_address" {
    null = false
    type = varchar(255)
  }
  column "payment_pending" {
    null = false
    type = varchar(100)
  }
  column "active_user" {
    null = false
    type = varchar(255)
  }
  primary_key {
    columns = [column.id]
  }
}
schema "public" {
}
```

> Now, lets add the following index definition to the file:

Reviewer: ```suggestion
Now, let's add the following index definition to the file:
```
